htdocs/accountancy/bookkeeping/export.php

```diff
@@ -560,12 +556,12 @@
 			$sqlwhere[] = "t.subledger_account >= '".$db->escape($value)."'";
 		} elseif ($key == 't.subledger_account<=') {
 			$sqlwhere[] = "t.subledger_account <= '".$db->escape($value)."'";
-		} elseif ($key == 't.fk_doc' || $key == 't.fk_docdet' || $key == 't.piece_num') {
-			$sqlwhere[] = $db->sanitize($key).'='.((int) $value);
+			// } elseif ($key == 't.fk_doc' || $key == 't.fk_docdet' || $key == 't.piece_num') { // these fields doesn't exists
+			// 	$sqlwhere[] = $db->sanitize($key).'='.((int) $value);
 		} elseif ($key == 't.subledger_account' || $key == 't.numero_compte') {
 			$sqlwhere[] = $db->sanitize($key)." LIKE '".$db->escape($db->escapeforlike($value))."%'";
-		} elseif ($key == 't.subledger_account') {
-			$sqlwhere[] = natural_search($key, $value, 0, 1);
+			// } elseif ($key == 't.subledger_account') { // this code is unreachable, test is always false
```

Comment:
Are you sure
Lien 386, we have
$filter['t.subledger_account'] = ...

Reply:
see test 
![image](https://github.com/user-attachments/assets/9b7fa9c9-9513-48b8-b7e7-daee1cb9df2a)